datalake-spark3/src/main/scala/bio/ferlab/datalake/spark3/utils/ClassGenerator.scala

```diff
@@ -143,17 +146,44 @@ case class $className(${fields.mkString("", s",\n${spacing.mkString}" , ")")}"""
        |""".stripMargin
   }
 
-  def writeCLassFile(packageName: String,
-                     className: String,
-                     df: DataFrame,
-                     rootFolder: String = getClass.getResource(".").getFile): Unit = {
+  def getRandomValue(dataType: org.apache.spark.sql.types.DataType): Any = dataType match {
+    case _: org.apache.spark.sql.types.StringType => Random.alphanumeric.take(10).mkString
+    case _: org.apache.spark.sql.types.IntegerType => Math.abs(Random.nextInt())
+    case _: org.apache.spark.sql.types.DoubleType => Math.abs(Random.nextDouble())
+    case _: org.apache.spark.sql.types.BooleanType => Random.nextBoolean()
+    case _: org.apache.spark.sql.types.DateType => java.sql.Date.valueOf("2022-02-20")
+    case _: org.apache.spark.sql.types.TimestampType => java.sql.Timestamp.valueOf("2022-02-20 12:34:56.789")
+    case _: org.apache.spark.sql.types.DecimalType => BigDecimal.valueOf(Math.abs(Random.nextInt()))
+    case _: org.apache.spark.sql.types.LongType => Math.abs(Random.nextLong())
+    case _: org.apache.spark.sql.types.ShortType => Math.abs(Random.nextInt().toShort)
+    case _ => null
+  }
+
+  def writeCLassFile(
+                      packageName: String,
+                      className: String,
+                      df: DataFrame,
+                      rootFolder: String = getClass.getResource(".").getFile,
+                      randomizeValues: Boolean = false
+                    ): Unit = {
+    val dfToWrite = if (randomizeValues) {
+      val schema = df.schema
+      val randomData = (1 to 2).map { _ =>
```

Comment:
Pas besoin d'utiliser le `to` si on génère juste une Row. Si tu génères une Seq avec juste une Row tu peux le passer à parallelize je pense.
```scala
val randomData = Seq(Row(schema.fields.map { field => ...
```

Reply:
done modifié pour que ca crée un df avec 1 ligne seulement